iml-config/src/nginx.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlConfigError;
+use console::Term;
+use std::{env, str};
+use structopt::StructOpt;
+use tokio::fs;
+
+#[derive(Debug, StructOpt)]
+pub enum NginxCommand {
+    /// Generate Nginx conf
+    #[structopt(name = "generate-config")]
+    GenerateConfig {
+        /// Set the nginx config path
+        #[structopt(short = "p", long = "path")]
+        template_path: String,
+
+        #[structopt(short = "o", long = "output")]
+        output_path: Option<String>,
+    },
+}
+
+const TEMPLATE_VARS: &[&str] = &[
+    "REPO_PATH",
+    "HTTP_FRONTEND_PORT",
+    "HTTPS_FRONTEND_PORT",
+    "HTTP_AGENT_PROXY_PASS",
+    "HTTP_AGENT2_PROXY_PASS",
+    "HTTP_API_PROXY_PASS",
+    "IML_API_PROXY_PASS",
+    "WARP_DRIVE_PROXY_PASS",
+    "MAILBOX_PATH",
+    "MAILBOX_PROXY_PASS",
+    "SSL_PATH",
+    "DEVICE_AGGREGATOR_PORT",
+    "DEVICE_AGGREGATOR_PROXY_PASS",
+    "UPDATE_HANDLER_PROXY_PASS",
+    "GRAFANA_PORT",
+    "GRAFANA_PROXY_PASS",
+    "INFLUXDB_PROXY_PASS",
+    "TIMER_PROXY_PASS",
+    "INCLUDES",
+];
+
+fn replace_template_variables(contents: &str, template_vars: &[&str]) -> String {
+    let mut updated_contents: String = contents.to_string();
+    for var in template_vars {
+        updated_contents = updated_contents.replace(
+            // find a better way to do this
+            &format!("{{{{{}}}}}", var),
+            &env::var(var).unwrap_or_else(|_| panic!("{} variable not set", var)),
```

Comment:
There could be a new variable in the template, but not in this list (e .g. a typo).

Reply:
https://github.com/nickel-org/rust-mustache ?

Reply:
I don't think we need to add a new lib for this.

Reply:
Now that i've got this in i'm looking for a better way to do this replacement.

Reply:
I am thinking about this sequence: encounter a var ref in template -> find this var in environment (fail if not found) -> substitute.

Reply:
> I am thinking about this sequence: encounter a var ref in template -> find this var in environment (fail if not found) -> substitute.

Yes, that sounds like a good approach. It could be done by reading line by line and stopping on an Error.